workshop/content/intermediate/templates/dynamic-references/_index.md

```diff
@@ -0,0 +1,249 @@
+
+When you use a dynamic reference, CloudFormation retrieves the value of the specified reference when necessary during stack and [change set](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-cfn-updating-stacks-changesets.html) operations. However, CloudFormation never stores the actual reference value.
+
+### Topics Covered
+
+By the end of this lab, you will be able to:
+
+* Compose a *dynamic reference string* to access an external value in your CloudFormation template.
+* Retrieve a specific version, or the *latest* version of a Parameter Store parameter.
+* Retrieve a specific version of a Secrets Manager secret.
+    * Extract a value for a specific key, from a secret that uses a JSON data format.
+
+### Start Lab
+
+#### Dynamic References for Parameter Store
+
+Consider a scenario where you are required to provide life cycle environments for your development teams. This practice often involves practices that include building and distributing custom [Amazon Machine Images](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/AMIs.html) (AMIs) with latest operating system updates, hardening requirements, and third-party software agents you require.
+
+After you (or a team in your organization) builds a custom AMI, you can choose to use Parameter Store to store the identifier of the AMI. This makes it easier for you to programmatically point to an AMI you wish to use when you launch EC2 instances, thus reducing the likelihood of configuration mistakes.
+
+In this lab, you will create a Parameter Store parameter to persist an AMI ID: instead of a custom AMI, you will use the latest _Amazon Linux 2 AMI, 64-bit x86_ available in a region of your choice. You will then reference your parameter in the `ImageId` [property](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-instance.html#cfn-ec2-instance-imageid) of an EC2 instance that you describe in a template.
+
+Let’s get started! Choose to follow steps shown next:
+
+1. Navigate to the the Amazon EC2 [Console](https://console.aws.amazon.com/ec2/v2/home?region=us-east-1#LaunchInstanceWizard:), and [choose the Region](https://docs.aws.amazon.com/awsconsolehelpdocs/latest/gsg/select-region.html) you wish to use. Next, locate the latest *Amazon Linux 2 AMI, (64-bit x86)*, and note the AMI ID (e.g., `ami-abcd1234`). You will use this value in the next step.
+
+![ec2](dynamic-references/ec2-console-ami-picker.png)
```

Comment:
Although this is what I thought about initially. I'm not inclined to guiding the customer to fetch the AMI ID using the CLI because the tutorial is about using Dynamic References for SSM and if I were to ask the customer to retrieve a value from SSM using the CLI it would be counter-intuitive to the benefits of using dynamic references. Please let me know what you think about this ?

Reply:
Fair enough. This was just a small suggestion, so the customers don't need to use multiple steps to get `ami-id` and not to jump between console and terminal.  But your way is absolutely valid so we can leave it a sit is.